File: 2610_matrix/matrix.py
from typing import List


class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        res = []

        j = 0
        k = 0

        pov = [nums]
        while k < len(nums):
            subarr = []
            p = []

            for i in pov[j]:
                if i not in subarr:
                    subarr.append(i)
                    k += 1
                else:
                    p.append(i)
            j += 1
            pov.append(p)
            res.append(subarr)

        return res

File: 2610_matrix/matrix.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var findMatrix = function (nums) {
  const freq = Array(nums.length + 1).fill(0);
  const res = [];

  nums.forEach((i) => {
    if (freq[i] >= res.length) {
      res.push([]);
    }

    res[freq[i]].push(i);
    freq[i]++;
  });

  return res;
};
